docs/virtual_machines/istio_service_mesh.md

```diff
@@ -0,0 +1,144 @@
+# Istio service mesh
+
+Service mesh allows to monitor, visualize and control traffic between microservices.
+Kubevirt supports running VMs as a part of Istio service mesh.
+
+## Limitations
+
+- Istio service mesh is only supported with a pod network masquerade binding.
+
+- Istio uses a [list of ports](https://istio.io/latest/docs/ops/deployment/requirements/#ports-used-by-istio) for its own purposes, these ports must not be explicitly specified in a VMI interface.
+
+## Prerequisites
+
+- This guide assumes that Istio is already deployed and uses Istio CNI Plugin. See [Istio documentation](https://istio.io/latest/docs/) for more information.
+
+- The target namespace where the VM is created must be labelled with `istio-injection=enabled` label.
+
+- If Multus is used to manage CNI, the following `NetworkAttachmentDefinition` is required in the application namespace:
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+```
+
+## Create a VirtualMachineInstance with enabled Istio proxy injecton
+
+The example below specifies a VMI with masquerade network interface and `sidecar.istio.io/inject` annotation to register the VM to the service mesh. 
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  annotations:
+    sidecar.istio.io/inject: "true"
+  labels:
+    app: vmi-istio
+  name: vmi-istio
+spec:
+  domain:
+    devices:
+      interfaces:
+        - name: default
+          masquerade: {}
+      disks:
+        - disk:
+            bus: virtio
+          name: containerdisk
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+    - name: default
+      pod: {}
+  terminationGracePeriodSeconds: 0
+  volumes:
+    - name: containerdisk
+      containerDisk:
+        image: registry:5000/kubevirt/fedora-cloud-container-disk-demo:devel
+```
+
+Istio expects each application to be associated with at least one Kubernetes service. Create the following Service exposing port 8080:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmi-istio
+spec:
+  selector:
+    app: vmi-istio
+  ports:
+    - port: 8080
+      name: http
+      protocol: TCP
+```
+
+**Note:** The `sidecar.istio.io/inject` must always be provided when using Istio service mesh to instruct KubeVirt to perform
+required networking configuration.
```

Comment:
You've already mentioned this twice - once in the `Prerequisites` section, another on the "Create a VirtualMachineInstance with enabled Istio proxy injecton". I would remove it from the latter, or here.

Reply:
In prerequisites section I'm mentioning the namespace label, this is a VMI annotation, it's a different thing.

To explaing this more broadly, I'm mentioning it in the note on purpose, because for regular pods, having the annotation is not required, labelled namespace is enough to get the sidecar proxy injected. We use the annotation to perform special configuration, which is why we always need it.
(This feature gap doesn't exist on d/s because d/s service mesh also requires each workload to be annotated explicitly)

Do you still think that having the Note is unnecessary?

Reply:
Oh I see. That changes the picture.

Would you make this note a bit more specific ? Mention "Each Istio aware VMI must feature the ...." or something like that. 

I honestly missed the point this was on the VMI.

Reply:
Yep, your suggestion sounds more clearly. Updated